core/src/com/mygdx/game/models/Session.java

```diff
@@ -0,0 +1,209 @@
+package com.mygdx.game.models;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Random;
+
+/**
+ * The Session model keeps track of players, selected brains, total brains and
+ * rounds.
+ *
+ * This class implements the MVC pattern.
+ * The parameters given in the constructor makes it easier to implement the quality requirement M1.
+ */
+
+public class Session {
+    public static final int MAX = 9999;
+    public static final int MIN = 1000;
+
+    private final int maxHitPoints;
+    private final int brainDamage;
+    private final int maxSelectedBrains;
+    private final int maxRounds;
+    private final ArrayList<Player> players;
+    private ArrayList<Brain> selectedBrains;
+    private ArrayList<Brain> allBrains;
+    private ArrayList<Round> rounds;
+    private ArrayList<Round> currentRounds;
+    private boolean activeRound;
+    private final int sessionCode;
+
+    /**
+     * The Constructor for a singleplayer session
+     * @param maxHitPoints: The max number of hit points a wall has
+     * @param brainDamage: How much damage each brain does
+     * @param maxSelectedBrains: The mac number of brains that can be selected in each
+     *                         elimination phase
+     * @param maxRounds: The max number of rounds that can be played
+     * @param player: Which player is playing the session
+     */
+    public Session(int maxHitPoints, int brainDamage, int maxSelectedBrains, int maxRounds,
+                   Player player) {
+        this(maxHitPoints, brainDamage, maxSelectedBrains, maxRounds,
+                new ArrayList<>(Collections.singletonList(player)), generateSessionCode());
+    }
+
+    /**
+     * The Constructor for a multiplayer session
+     * @param maxHitPoints: The max number of hit points a wall has
+     * @param brainDamage: How much damage each brain does
+     * @param maxSelectedBrains: The mac number of brains that can be selected in each
+     *                         elimination phase
+     * @param maxRounds: The max number of rounds that can be played
+     * @param players: The list of players that
+     */
+    public Session(int maxHitPoints, int brainDamage, int maxSelectedBrains, int maxRounds,
```

Comment:
How do you generate the first session code if the constructer needs it?

Reply:
There is a public static method generateSessionCode()